spec/features/admin_node_update_feature_spec.rb

```diff
@@ -40,27 +40,25 @@
       expect(page).to have_content("Reboot to update")
     end
 
+    # rubocop:disable RSpec/ExampleLength
     scenario "User clicks on 'Reboot to update'", js: true do
       allow(::Velum::Salt).to receive(:call).and_return(true)
 
       # clicks on "Update admin node"
+      expect(page).to have_selector(".update-admin-btn", visible: true)
       find(".update-admin-btn").click
 
       # wait modal to appear
       expect(page).to have_content("Reboot to update")
 
       # clicks on "Reboot to update"
+      expect(page).to have_selector(".reboot-update-btn", visible: true)
       find(".reboot-update-btn").click
 
-      expect(page).to have_content("Rebooting...", wait: 10)
-
-      # NOTE: The check below is flaky: sometimes it passes and sometimes it
-      # doesn't. We believe that this is Capybara/Poltergeist to blame. Since we
-      # already have this check in `spec/controllers/updates_controller_spec.rb`
-      # and getting the "Rebooting..." flashy message already guarantees that
-      # the controller was reached, we have simply commented out the line below.
-      # expect(::Velum::Salt).to have_received(:call).once
+      expect(page).to have_content("Rebooting...")
+      expect(::Velum::Salt).to have_received(:call).once
```

Comment:
This is why I placed the `WIP` back. I suspect there is a real problem here, and I have seen this working (is not always a failure). This is the only flaky test left, giving it some thought, hope to have it fixed soon...